src/dft/backends/mklcpu/commit.cpp

```diff
@@ -96,44 +97,137 @@ class commit_derived_impl : public detail::commit_impl {
     }
 
     template <typename... Args>
-    DFT_ERROR set_value_item(DFTI_DESCRIPTOR_HANDLE hand, enum DFTI_CONFIG_PARAM name,
-                             Args... args) {
-        DFT_ERROR value_err = DFT_NOTSET;
-        value_err = DftiSetValue(hand, name, args...);
-        if (value_err != DFTI_NO_ERROR) {
-            throw oneapi::mkl::exception("dft", "set_value_item", std::to_string(name));
+    void set_value_item(DFTI_DESCRIPTOR_HANDLE hand, enum DFTI_CONFIG_PARAM name, Args... args) {
+        if (auto ret = DftiSetValue(hand, name, args...); ret != DFTI_NO_ERROR) {
+            throw oneapi::mkl::exception(
+                "dft/backends/mklcpu", "set_value_item",
+                "name: " + std::to_string(name) + " error: " + std::to_string(ret));
         }
-
-        return value_err;
     }
 
-    void set_value(DFTI_DESCRIPTOR_HANDLE& descHandle, detail::dft_values<prec, dom> config) {
-        set_value_item(descHandle, DFTI_INPUT_STRIDES, config.input_strides.data());
-        set_value_item(descHandle, DFTI_OUTPUT_STRIDES, config.output_strides.data());
-        set_value_item(descHandle, DFTI_BACKWARD_SCALE, config.bwd_scale);
-        set_value_item(descHandle, DFTI_FORWARD_SCALE, config.fwd_scale);
-        set_value_item(descHandle, DFTI_NUMBER_OF_TRANSFORMS, config.number_of_transforms);
-        set_value_item(descHandle, DFTI_INPUT_DISTANCE, config.fwd_dist);
-        set_value_item(descHandle, DFTI_OUTPUT_DISTANCE, config.bwd_dist);
-        set_value_item(
-            descHandle, DFTI_PLACEMENT,
-            (config.placement == config_value::INPLACE) ? DFTI_INPLACE : DFTI_NOT_INPLACE);
+    void set_value(DFTI_DESCRIPTOR_HANDLE& descHandle, detail::dft_values<prec, dom>& config) {
```

Comment:
Can you give some explanations why is this change needed? I am wondering if this is a generic improvement (and if so if it needs to go in the main implementation PR) or if this is needed to allow recommitting?

Reply:
This is needed so that `config.changed.clear()` is communicated back to the `dft_values`

Reply:
unless you are referring to `set_value_item`, in that case it is general improvement since I noticed the return wasn't really used. I will add it to the main PR.

Reply:
That wasn't clear, I was referring to all the changes in `set_value` and in particular why we were adding the for loop and switch. It looks like this won't be needed if we remove `changed` so we're good.
I agree for changing the return type to `void` and adding the reference. By the way maybe `config` can be a const reference?

Reply:
Ah ok. I've removed the changes to `set_value` and made `config` a const reference.